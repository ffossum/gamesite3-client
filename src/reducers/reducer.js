/* @flow */
import type { Action } from '../actions/actionTypes';

export type State = {};
const initialState = {};
export default function reducer(state?: State = initialState, action: Action) {
  switch (action.type) {
    default:
      return state;
  }
}
