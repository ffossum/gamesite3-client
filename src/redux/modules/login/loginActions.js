/* @flow */
export const LOGIN_REQUEST = 'login/request';
export const LOGIN_SUCCESS = 'login/success';
export const LOGIN_FAILURE = 'login/failure';

import type { SessionUser } from '../session/sessionReducer';

export type Login = {
  email: string,
  password: string,
};
export type LoginRequestAction = {
  type: typeof LOGIN_REQUEST,
  payload: Login,
};
export function loginRequest(login: Login): LoginRequestAction {
  return {
    type: LOGIN_REQUEST,
    payload: login,
  };
}

type LoginSuccessAction = {
  type: typeof LOGIN_SUCCESS,
  payload: SessionUser,
};
export function loginSuccess(user: SessionUser): LoginSuccessAction {
  return {
    type: LOGIN_SUCCESS,
    payload: user,
  };
}

type LoginFailureAction = {
  type: typeof LOGIN_FAILURE,
};
export function loginFailure(): LoginFailureAction {
  return {
    type: LOGIN_FAILURE,
  };
}

export type LoginAction =
  | LoginRequestAction
  | LoginSuccessAction
  | LoginFailureAction;
