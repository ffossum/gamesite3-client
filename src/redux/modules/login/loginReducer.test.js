/* @flow */
/* eslint-env jest */
import reducer from './loginReducer';
import { loginRequest, loginSuccess, loginFailure } from './loginActions';

describe('login reducer', () => {
  const login = {
    email: 'bob@test.com',
    password: 'bobisthebest',
  };
  const initialState = reducer(undefined, { type: '@@INIT' });
  test('initial state', () => {
    expect(initialState).toMatchSnapshot();
  });

  test('is loading after request', () => {
    const action = loginRequest(login);
    const state = reducer(initialState, action);
    expect(state.loading).toBe(true);
  });

  test('is no longer loading after success', () => {
    let state = reducer(initialState, loginRequest(login));
    state = reducer(
      state,
      loginSuccess({
        email: 'bob@test.com',
        id: 'abc123',
        username: 'bob',
      }),
    );
    expect(state.loading).toBe(false);
  });

  test('is no longer loading after failure', () => {
    let state = reducer(initialState, loginRequest(login));
    state = reducer(state, loginFailure());
    expect(state.loading).toBe(false);
  });
});
