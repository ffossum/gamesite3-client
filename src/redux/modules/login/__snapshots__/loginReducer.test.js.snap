// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login reducer initial state 1`] = `
Object {
  "loading": false,
}
`;
